### ISSUE TITLE
Suggestion: Component (and design) __len__ and __getitem__

### ISSUE THREAD

**Issue by akaszynski** [ansys/pyansys-geometry#152]
Would be great if I could `len(mycomp)` and get a real length. Likewise with accessing bodies via a body name or even ID. For example, if I create a body within the component named 'my_body', I should be able to access it via `mycomp['my_body']`. We can even consider supporting an index based `__getitem`.

If component contains more than just bodies, then there's a good reason to move this logic into a `Bodies` class and expose that from `Component.bodies`.

**Comment by jmabshof** [comment]
@RobPasMue still relevant?

**Comment by RobPasMue** [comment]
Yeah - this is already somewhat available, also the way the info is accessed changed a bit. Closing.
